cardinal/config.go

```diff
@@ -1,49 +1,48 @@
 package cardinal
 
 import (
-	"os"
+	"github.com/JeremyLoy/config"
+)
+
+type RunMode string
 
-	"github.com/rs/zerolog/log"
+const (
+	RunModeProd RunMode = "production"
+	RunModeDev  RunMode = "development"
 )
 
 const (
-	ModeProd             = "production"
-	ModeDev              = "development"
-	DefaultMode          = ModeDev
-	DefaultNamespace     = "world-1"
-	DefaultRedisPassword = ""
-	DefaultRedisAddress  = "localhost:6379"
+	DefaultRunMode                   = RunModeDev
+	DefaultNamespace                 = "world-1"
+	DefaultRedisPassword             = ""
+	DefaultRedisAddress              = "localhost:6379"
+	DefaultBaseShardSequencerAddress = ""
+	DefaultBaseShardQueryAddress     = ""
 )
 
 type WorldConfig struct {
-	RedisAddress      string
-	RedisPassword     string
-	CardinalNamespace string
-	CardinalPort      string
```

Comment:
i removed port cause it wasn't actually being used anywhere.

Reply:
in cardinal/server/server.go the CARDINAL_PORT env variable is used to set the port. For consistency, I'd suggesting keeping CARDINAL_PORT here, and passing in the port to that package.

If that's not trivial to do here, you should make a linear ticket to capture that work.

Reply:
ill make a new ticket

Reply:
https://linear.app/arguslabs/issue/WORLD-689/remove-automatic-port-instantiation-in-cardinalserver